packages/block-library/src/cover/edit.js

```diff
@@ -311,11 +314,36 @@ function CoverEdit( {
 		minHeight,
 		minHeightUnit,
 		style: styleAttribute,
-		url,
 		alt,
 		allowedBlocks,
 		templateLock,
 	} = attributes;
+
+	const [ featuredImage ] = useEntityProp(
+		'postType',
+		postType,
+		'featured_media',
+		postId
+	);
+
+	const media = useSelect(
+		( select ) =>
+			featuredImage &&
+			select( coreStore ).getMedia( featuredImage, { context: 'view' } ),
+		[ featuredImage ]
+	);
+	const mediaUrl = media?.source_url;
+
+	// instead of destructuring the attributes
+	// we define the url and background type
+	// depending on the value of the useFeaturedImage flag
+	// to preview in edit the dynamic featured image
+	const url = useFeaturedImage && mediaUrl ? mediaUrl : attributes.url;
```

Comment:
Why are we waiting for mediaUrl to be filled to use it. I mean if "mediaUrl" is empty but "useFeaturedImage" is true, should we use the previous saved "url" or just show some kind of loader/placeholder? I think using the previous image could be misleading?

Reply:
We can use the block with the featured image while the post has no featured image, the url will be undefined. That is a state in which one gets if they start with a color cover and toggle featured image on. I think nothing will break by removing this extra check.

Reply:
Yes @youknowriad waiting on `mediaUrl` here was a future bug waiting to happen :D We're fine without it. As long as we're not setting these attributes we only care about them if `useFeaturedImage` is true.

PR updated.